git-sync-relay/gitsync.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+set -e
+
+GIT_SSH_COMMAND="ssh"
+
+if [ ! -d "$GIT_SYNC_ROOT" ]; then
+  echo "Error: GIT_SYNC_ROOT ${GIT_SYNC_ROOT} does not exist!"
+  exit 1
+fi
+
+if [ -n "$(ls -A "$GIT_SYNC_ROOT")" ]; then
+  echo "Error: GIT_SYNC_ROOT ${GIT_SYNC_ROOT} is not empty."
+  exit 1
+fi
+
+if [ -n "$GIT_SYNC_SSH" ] && [ "true" = "$GIT_SYNC_SSH" ]; then
+  GIT_SSH_COMMAND="$GIT_SSH_COMMAND -i $GIT_SSH_KEY_FILE"
+
+  if [ -n "$GIT_KNOWN_HOSTS" ] && [ "true" = "$GIT_KNOWN_HOSTS" ]; then
+    GIT_SSH_COMMAND="$GIT_SSH_COMMAND -o StrictHostKeyChecking=yes -o PasswordAuthentication=no -o KbdInteractiveAuthentication=no -o KbdInteractiveDevices=no -o UserKnownHostsFile=$GIT_SSH_KNOWN_HOSTS_FILE"
+  else
+    GIT_SSH_COMMAND="$GIT_SSH_COMMAND -o StrictHostKeyChecking=yes -o PasswordAuthentication=no -o KbdInteractiveAuthentication=no -o KbdInteractiveDevices=no"
```

Comment:
I made this suggestion in the wrong place. We should move this up to line 5.
```suggestion
    GIT_SSH_COMMAND="$GIT_SSH_COMMAND -o StrictHostKeyChecking=yes -o UserKnownHostsFile=$GIT_SSH_KNOWN_HOSTS_FILE"
  else
    GIT_SSH_COMMAND="$GIT_SSH_COMMAND -o StrictHostKeyChecking=yes"
```